DesignSafe-CI/portal
# Task/DES-1911 - Allow swapping between publication versions.

## Overview: ##
I still need to test publishing, amending, and versioning with these changes, but I'm sorting out an unrelated issue at the moment. The changes in this PR allow for the different versions of the publication to be swapped between as designed in Craig's mockups. The version selection is structured like so:

| ID                   | Version | Path |
| ------------- | --------|------|
| PRJ-1234 | original | `/published/PRJ-1234/file.txt` |
| PRJ-1234r1 | 1 | `/published/PRJ-1234r1/file.txt` |
| PRJ-1234r2 | 2 | `/published/PRJ-1234r2/file.txt` |
| PRJ-1234r... | ... | `/published/PRJ-1234r.../file.txt` |

If a version number is not provided when requesting a publication the "original" is always returned. The latest version number will also be returned within the publication object for reference.

## PR Status: ##


## Related Jira tickets: ##

* [DES-1911](https://jira.tacc.utexas.edu/browse/DES-1911)

## Summary of Changes: ##

- The regex patterns for the publication endpoints have been updated to account for a version number.
- Published Parent and Published View has been tweaked to account for version differences
- Minor UI changes

## Testing Steps: ##
1. If you've published several version of a project, navigate to the published area of the data depot and find that publication
2. When navigating into that publication you should be taken into the **original** version of that published project (this is going to be fixed with the published listing)
3. When in your published project should see a "Version" dropdown in the publication's metadata header. Clicking the dropdown should show all of the published versions and allow you to navigate between them.
4. Clicking the "Version Changes" next to the dropdown should show you the selected version, and the reason for the version's changes provided by the user when they created the new version of that publication. (This is not the project description)

## UI Photos:
![version-swap](https://user-images.githubusercontent.com/29575979/114631038-7ace8080-9c81-11eb-8ed3-a692596b089a.png)
